core/src/main/java/org/aerogear/mobile/core/configuration/MobileCoreJsonParser.java

```diff
@@ -19,12 +20,17 @@
  */
 public class MobileCoreJsonParser {
 
-    private final Map<String, ServiceConfiguration> values = new TreeMap<>();
+    private static final Map<String, ServiceConfiguration> values = new TreeMap<>();
+    private static final Map<String, String> hashes = new HashMap<>();
+
 
     private MobileCoreJsonParser(final InputStream jsonStream) throws IOException, JSONException {
```

Comment:
Parser should have a `parse` method that returns a parse result.
If the parsing is done inside the constructor, then this is not a parser, but an object that is built from a InputStream.